### PR TITLE
[10-10CG] Delete pdf in ensure block on download_pdf

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- This work is behind a feature toggle (flipper): YES
- Currently, in the `download_pdf` method, we generate a pdf, load it into a variable, delete the pdf file, and then `send_data` to send the variable to front end. This change will call the `send_data` and then in an ensure block delete the pdf file, hopefully eliminating any potential race conditions. 
- This change is one potential way to make the endpoint more resilient, as we have seen a few errors where there is an issue accessing the pdf file. 
- 10-10 Health Apps
- `caregiver1010`

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/93581

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
- 10-10CG

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: spec/requests/v0/caregivers_assistance_claims_spec.rb [hunk 1]
```diff
@@ -16,29 +16,37 @@
...
+    let(:form_data) { get_fixture('pdf_fill/10-10CG/simple').to_json }
+    let(:claim) { build(:caregivers_assistance_claim, form: form_data) }
+    let(:body) { { caregivers_assistance_claim: { form: form_data } }.to_json }
+
     after do
       FileUtils.rm_f(response_pdf)
     end
 
     it 'returns a completed PDF', run_at: '2017-07-25 00:00:00 -0400' do
-      form_data = get_fixture('pdf_fill/10-10CG/simple').to_json
```
Comment (coope93) [inline spec/requests/v0/caregivers_assistance_claims_spec.rb:28]: Moved these to lets to DRY this up a bit.

**Comment by stevenjcumming** [comment]
@coope93 please ask a teammate for a review, then I'll review it

File: spec/requests/v0/caregivers_assistance_claims_spec.rb [hunk 2]
```diff
@@ -16,29 +16,37 @@
   let(:get_schema) { -> { VetsJsonSchema::SCHEMAS['10-10CG'].clone } }
 
   describe 'POST /v0/caregivers_assistance_claims/download_pdf' do
+    subject do
+      post('/v0/caregivers_assistance_claims/download_pdf', params: body, headers:)
+    end
+
+    before do
+      allow(Flipper).to receive(:enabled?).with(:caregiver1010).and_return(false)
+    end
```
Comment (Mitch-A6) [inline spec/requests/v0/caregivers_assistance_claims_spec.rb]: recommend wrapping this in a context around the (sole) spec that relies on it, so we're not restubbing it in :67.

**Review by Mitch-A6 — APPROVED** [review]
👍

**Review by stevenjcumming — APPROVED** [review]

**Review by ericboehs — APPROVED** [review]

**Pull request merged**